packages/compass-components/src/components/modal-title.tsx

```diff
@@ -9,7 +9,7 @@ const modalTitleStyles = css({
 });
 
 function ModalTitle(
-  props: React.ComponentProps<typeof H3>
+  props: Omit<React.ComponentProps<typeof H3>, 'as'>
```

Comment:
Not sure what the right way to do this would be. (Why would H3 even have as?)

Reply:
What error were you getting? It compiles fine for me when leaving it as `React.ComponentProps<typeof H3>` on top of this PR.

Reply:
not here but where ModalTitle gets used:

![Screenshot 2022-08-30 at 12 47 12](https://user-images.githubusercontent.com/69737/187428789-2d08cc20-7787-4d04-a81a-ee983cebf7fe.png)